docs/designs/2730-visibility.md

```diff
@@ -0,0 +1,293 @@
+critical when something goes wrong and someone else needs to take a look at the
+output to help troubleshoot the failure.
+
+* Cluster creation output should be captured to
+  `~/.config/tanzu/tkg/bootstrap-logs/${CLUSTER_NAME}.log`
+* The log should capture everything, regardless of verbosity level set by the
+  end user for console output
+* Additional debug command output, such as tailing CAPI controller logs on
+  failure, should be captured in the bootstrap log
+* To prevent hidden leaked filesystem space consumption, the bootstrap log
+  should be deleted as part of the `management-cluster delete` or `cluster
+  delete` operations
+
+#### FR4 - Debug capture on failed cluster deployment
+
+During deployment, we capture the kubectl configuration and other details about
+the bootstrap kind cluster and the cluster that is actively being deployed. We
+use some of this to help guide the user today:
+
+```txt
+Failure while deploying management cluster, Here are some steps to investigate the cause:
+Debug:
+kubectl get po,deploy,cluster,kubeadmcontrolplane,machine,machinedeployment -A --kubeconfig /home/test/.kube-tkg/tmp/config_6SDG31q1
+kubectl logs deployment.apps/ -n manager --kubeconfig /home/test/.kube-tkg/tmp/config_6SDG31q1
+```
+
+Since we have this information, we should actually perform some of the
+debugging steps for the end user.
+
+As a base requirement for this proposal, debug commands like getting pod and
+deployment status from the cluster should be run, with the output captured to
+the bootstrap log. We should also capture the error logs from the CAPI provider
+manager pod (assuming things have gotten that far) and include that in the
+bootstrap log.
+
+For the purposes of this design proposal, the requirement is only to capture
+this output so it can be used for later troubleshooting. But since we have this
+information, a future effort of building in some parsing and automatic triaging
+to the commands should be considered.
+
+### Implementation Details/Notes/Constraints
+
+* The main target for these changes are `management-cluster` and `cluster`
+  create commands
+* Output will be different between these two as there are different details
+  that are more relevant for one versus the other
+* The same output principles should be considered for things like `delete`
+  commands and any others that perform a multi-step process to accomplish
+
+### Security Model
+
+N/A
+
+### Risks and Mitigations
+
+Output is displayed on the console as well as the web UI, when used. This
+should be kept in mind during development to make sure the output display in
+the web UI is also useful and the changes do not introduce any UI breakage. The
+UI developers should also be included in reviews to make sure there are no
+concerns from their perspective on the way changes may impact the UI.
+
+## Compatibility
+
+N/A
+
+## Alternatives
+
+An alternative - and really the current approach - would be to leave the output
```

Comment:
Just a general comment. Another alternative is tooling to grab relevant "stuff"... but that is also proving to be equally difficult to ramp up on. Maybe another proposal to implement better tools.